CI Visibility - Swift - Add information about the GitHub action (#16699)
DataDog/documentation: content/en/continuous_integration/tests/swift.md

[1]: https://github.com/DataDog/dd-sdk-swift-testing/releases
{{% /tab %}}
{{% tab "GitHub Actions" %}}

If you use GitHub, you can use the [Swift test action][1] from GitHub marketplace to automatically configure and run your tests. By default, the rest of the configuration explained in this document can be skipped (except the configuration of the action itself), but can also use the configuration environment variables for disabling or configuring extra functionality.

> This option has the benefit that no code changes are needed but has the drawback that it is less flexible to configure and run.

Author: * It refers to the Swift test actions
* It is less flexible than he other ways, yes

> This option has the benefit that no code changes are needed but has the drawback that it is less flexible to configure and run.

Reviewer: ```suggestion
Compared to other methods, such as Cocoapods and Framework linking, the Swift test action option may be less flexible to configure and run, but requires no code changes.
```

[1]: https://github.com/DataDog/dd-sdk-swift-testing/releases
{{% /tab %}}

> {{% /tab %}}

Reviewer 2: ```suggestion
```

{{% tab "GitHub Actions" %}}

> If you use GitHub, you can use the [Swift Test Action][1] from GitHub marketplace to automatically configure and run your tests. By default, the rest of configuration explained in this document can be skipped (except the configuration of the action itself), but can also use the configuration environment variables for disabling or configuring extra functionality.

Reviewer 3: ```suggestion
If you use GitHub, you can use the [Swift test action][1] from GitHub marketplace to automatically configure and run your tests. By default, the rest of the configuration explained in this document can be skipped (except the configuration of the action itself), but can also use the configuration environment variables for disabling or configuring extra functionality.
```

Always call `module.end()` at the end so that all the test info is flushed to Datadog.

## Best practices

Follow these practices to take full advantage of the testing framework and CI Visibility.

### Generate symbols file when building

> Build in Xcode using `DWARF with dSYM File` (or `-Xswiftc -debug-info-format=dwarf` if building with `swift`)

Author: ```suggestion
Build your code in Xcode using `DWARF with dSYM File` (or `-Xswiftc -debug-info-format=dwarf` if building with `swift`)
```

> The testing framework uses symbol files for some of its functionality: simbolicating crashes, reporting test source location, report codeowners... It automatically generates the symbol file when debug symbols are embedded in the binaries, but it can take some extra time at loading.

Reviewer 3: I believe this line needs a further edit: "and report codeowners" should be "and reporting code owners."

### Disable sandbox for UI Tests on macOS

> With some Xcode versions UITest bundles are built by default with sandbox settings, the testing framework needs to run some system commands with `xcrun` and this settings dissallow running them, so must be disabled.

Reviewer: ```suggestion
In some Xcode versions, UI Test bundles are built with a sandbox by default. The settings that come with a sandbox prevent the testing framework from being run by some system commands with `xcrun`, so you'll need to disable it.
```

> Sandbox can be disabled by adding Entitlements to the UITest runner bundle & adding `App Sandbox = NO` to them. You can also create a .entitlement file and add it to the Signing Build Settings, this file should should include the following content:

Reviewer: ```suggestion
Disable the sandbox by adding Entitlements to the UI Test runner bundle, then adding `App Sandbox = NO` to them. You can also create an `.entitlement` file and add it to the Signing Build Settings. This file should should include the following content:
```
